docs/API/API-functions.md

```diff
@@ -358,27 +358,74 @@ int srt_bind(SRTSOCKET u, const struct sockaddr* name, int namelen);
 
 Binds a socket to a local address and port. Binding specifies the local network
 interface and the UDP port number to be used for the socket. When the local 
-address is a form of `INADDR_ANY`, then it's bound to all interfaces. When the 
-port number is 0, then the port number will be system-allocated if necessary.
+address is a wildcard (`INADDR_ANY` for IPv4 or `in6addr_any` for IPv6), then
+it's bound to all interfaces (although see `SRTO_IPV6ONLY` and an additional
+information below for details about the wildcard address in IPv6).
+
+Binding is necessary for every socket to be used for communication. It can be
+bound implicitly, when the socket is being connected to a listening socket in
+the network using [`srt_connect`](#srt_connect) or a similar function that does
+the same, in which case the socket will be bound to a wildcard address of
+the IP version as specified in the `sockaddr::sa_family` field and port number
+0.  In other cases it must be bound explicitly using this function, or some
+other function that uses this functionality.
```

Comment:
```suggestion
Binding is necessary for every socket to be used for communication. It can be
bound implicitly, such as when the socket is being connected to a listening socket in
the network using [`srt_connect`](#srt_connect). Functions similar to this can bind 
a socket implicitly to a wildcard address of the IP version as specified in the 
`sockaddr::sa_family` field and port number 0.  In other cases, a socket must 
be bound explicitly using such functionality.
```

Reply:
What I meant here is that `srt_connect` is not the only function that initiates the connection process. Implicit binding is possible (meaning: the call to a binding function, such as this one - `srt_bind` - is not obligatory) only in case when the socket is used to initiate a connection to the other listener socket, which can be, most simply, accomplished by `srt_connect` or some other function that does the same thing, at least among other things that it might be doing (for example, `srt_connect_bind` performs the role of `srt_bind` function and then `srt_connect`). Maybe better to say something like "It can be bound implicitly, if the socket is to be used to initiate a connection to a listener socket, which can be done e.g. by `srt_connect` function"...

Reply:
How about this:
```suggestion
Binding is necessary for every socket to be used for communication. If the socket
is to be used to initiate a connection to a listener socket, which can be done,
for example, by the [`srt_connect`](#srt_connect) function, the socket is bound
implicitly to the wildcard address according to the IP family (`INADDR_ANY` for
`AF_INET` or `in6addr_any` for `AF_INET6`) and port number 0. In all other cases,
a socket must be bound explicitly by using the functionality of this function first.
```

Reply:
Above is okay. Here is another possibility based on your earlier comment:

```suggestion
Binding is necessary for every socket to be used for communication. 
Implicit binding (meaning the call to a binding function like `srt_bind` 
is not obligatory) is possible only when the socket is used to initiate 
a connection to a listener socket on the network. In this case it can 
be bound implicitly using the [`srt_connect`](#srt_connect) function, 
although `srt_connect` is not the only function that can initiate the 
connection process. This can be accomplished by other similar functions.
For example, `srt_connect_bind` performs the role of the `srt_bind` 
function and `srt_connect`, whereupon the socket will be bound to a 
wildcard address of the IP version as specified in the `sockaddr::sa_family` 
field and port number 0. In other cases, a socket must first  be bound explicitly 
by using similar functionality.
```